helper/dynamic_property_support.go

```diff
@@ -0,0 +1,85 @@
+package helper
+
+import (
+	"fmt"
+	"github.com/buildpacks/libcnb"
+	"github.com/magiconair/properties"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"os"
+	"strings"
+)
+
+const prefix = "BPL_TOMCAT_ENV_"
+
+type DynamicPropertySupport struct {
+	Bindings libcnb.Bindings
+	Logger   bard.Logger
+}
+
+func (a DynamicPropertySupport) Execute() (map[string]string, error) {
+	systemProperties := make(map[string]string)
+
+	for _, b := range a.Bindings {
+		a.Logger.Infof("Configuring properties for binding %s", b.Name)
+
+		for k, v := range b.Secret {
+			systemProperties[fmt.Sprintf("%s.%s", b.Name, k)] = v
+		}
+	}
+
+	// lets get all environment variables named BPL_TOMCAT_ENV_*
+	envVariables := os.Environ()
+	for _, envVariable := range envVariables {
+		parts := strings.Split(envVariable, "=")
+		if strings.HasPrefix(parts[0], prefix) {
+			a.Logger.Infof("Configuring property from env var %s", parts[0])
+			value := os.Getenv(parts[0])
+			systemProperties[toSystemProperty(parts[0])] = value
+		}
```

Comment:
I'm curious about this. 

so `os.Environ()` will give you a list of all the env variables. So we'll get something like:

```
BPL_TOMCAT_ENV_FOO1=bar1
BPL_TOMCAT_ENV_FOO2=bar2
...
```

Then we split that to get `parts`, where parts[0] is the key and parts[1] is the value.

Then we filter out anything that's not `BPL_TOMCAT_ENV_*`. 

What I don't get is the next part, why are we calling `os.Getenv(parts[0])`? That would then go and attempt to fetch the value from another env variable. From the example, `parts[0] == "BPL_TOMCAT_ENV_FOO1"` and `parts[1] == "bar1"`. We should be able to just use `parts[1]`, right? Or am I missing something?

I think you just want to use `parts := strings.SplitN(env, "=", 2)` so that parts is always only length two. That way something like `BPL_TOMCAT_ENV_FOO=bar=baz` doesn't break things.

Reply:
I was using this to get around the fact that we may have entries like `BPL_TOMCAT_ENV_FOO=bar=baz`, `strings.SplitN(env, "=", 2)` is a much cleaner approach.